server/src/main/java/org/opensearch/index/codec/fuzzy/IndexInputLongArray.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec.fuzzy;
+
+import org.apache.lucene.store.RandomAccessInput;
+import org.apache.lucene.util.RamUsageEstimator;
+import org.opensearch.OpenSearchException;
+import org.opensearch.common.util.LongArray;
+
+import java.io.IOException;
+
+/**
+ * A Long array backed by RandomAccessInput.
+ */
+class IndexInputLongArray implements LongArray {
+
+    private final RandomAccessInput input;
+    private final long size;
+
+    IndexInputLongArray(long size, RandomAccessInput input) {
+        this.size = size;
+        this.input = input;
+    }
+
+    @Override
+    public void close() {}
+
+    @Override
+    public long size() {
+        return size;
+    }
+
+    @Override
+    public synchronized long get(long index) {
+        try {
+            return input.readLong(index << 3);
```

Comment:
Wanted to double check on this usage, Since IndexInput are not supposed to be shard between threads, we should call this our explicitly in the javadocs atleast (this works because we are doing a random read, it may not work if these were sequential reads)